plugins/out_s3/s3.c

```diff
@@ -42,6 +43,16 @@
 #define DEFAULT_S3_PORT 443
 #define DEFAULT_S3_INSECURE_PORT 80
 
+#define s3_retry_error(ctx, fmt, ...)                                    \
+    if (flb_log_check_level(ctx->log_level, FLB_LOG_ERROR))             \
+        flb_log_print(FLB_LOG_ERROR, NULL, 0, "[output:%s:%s] " fmt,    \
+                      ctx->p->name, flb_output_name(ctx), ##__VA_ARGS__)
+
+#define s3_retry_warn(ctx, fmt, ...)                                     \
+    if (flb_log_check_level(ctx->log_level, FLB_LOG_WARN))              \
+        flb_log_print(FLB_LOG_WARN, NULL, 0, "[output:%s:%s] " fmt,     \
+                      ctx->p->name, flb_output_name(ctx), ##__VA_ARGS__)
+
```

Comment:
@matthewfala What do you think of this?

I think we should not be using the flb_log_print function in our plugin code, its sort of like using a private internal function for a library instead of the public function. 

Let's use the functions that the library intends us to use- which are `flb_plg_warn` and etc.

Reply:
I don't see why this would be needed, and don't really like this.